### PR TITLE
Search trucks

### PR DESCRIPTION
**Pull request by AlejandroNavarro1**
This is a basic component for searching up trucks.

### CODE REVIEW

File: food-truck-frontend/src/components/search/SearchRow.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            </div>
+        )
+    }
+
+    return(
+        <div>
+            <Divider/>
+            <ListItem button onClick={() => {router.push(`/truck/${truck.truck.id}`)}}>
+                <ListItemText primary={truck.truck.name}/>
+                <ListItemText secondary={truck.truck.description ? truck.truck.description : "No Description"}/>
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/search/SearchRow.tsx:29]: Maybe adding a button to visit the truck's page would be nice. Not needed now, though.

File: food-truck-api/src/main/java/food/truck/api/truck/TruckRepository.java [hunk 2]
```diff
@@ -9,4 +9,5 @@
 public interface TruckRepository extends JpaRepository<Truck, Long> {
     List<Truck> findByName(String name);
     List<Truck> findByUserId(Long userId);
+    List<Truck> findByNameLikeOrDescriptionLike(String name, String description);
```
Comment (MicahSchiewe) [inline food-truck-api/src/main/java/food/truck/api/truck/TruckRepository.java:12]: I know there's an 'IgnoreCase' keyword you could apply to add case-insensitivity to the matching, but I'm not sure how to use it.

**Review by MicahSchiewe — APPROVED** [review]
LGTM

File: food-truck-frontend/src/pages/search/truck.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import {Button, Divider, FormGroup, Grid, List, ListItem, TextField} from "@material-ui/core";
+import api from "../../util/api";
+import SearchTruckRow from "../../components/search/SearchRow";
+import jssPluginPropsSort from "jss-plugin-props-sort";
+
+const truck = [{
+    id: -1,
+    name: "",
+    description: ""
+}]
```
Comment (KevinDeMars) [inline food-truck-frontend/src/pages/search/truck.tsx:11]: What is this for? If it's just to initialize this.state.trucks, why not initialize it to []?

**Review by KevinDeMars — APPROVED** [review]
Looks good

**Pull request merged**